shopfloor/components/scan_handler_product.py

```diff
@@ -15,7 +15,11 @@ class ProductHandler(Component):
     record_type = "product"
 
     def search(self, identifier):
-        return self._search.product_from_scan(identifier)
+        product = self._search.product_from_scan(identifier)
+        if not product:
+            packaging = self._search.packaging_from_scan(identifier)
+            product = packaging.product_id
```

Comment:
If the goal (not of this PR, but the general idea) is that all scenarios use this scan handler, we do not solve anything by scanning the product+packaging in the same place where some scenarios have a different flow regarding the packaging (like delivery).
We can keep it like this but we need to be aware of that.

Reply:
Answer from @simahawk in private: all good like this. Scan handler is used as a global feature on Shopfloor, it should not be used in the code of the specific scenario (for this, use `x_from_scan` methods)